src/main/java/use_cases/add_edit_expenses_use_case/ExpenseUCI.java

```diff
@@ -30,44 +32,42 @@ public static double toDouble(Object value) throws NumberFormatException {
 
     /**
      * Overrides method in ExpenseIB.
-     * Attempts to add an expense with information from ExpenseID and returns a ExpenseOD indicating whether fail/success after execution.
+     * Attempts to add an expense with information from ExpenseID and returns a ExpenseOD
+     * indicating whether fail/success after execution.
+     * ExpenseFactory methods are implemented to better adhere to Liskov's Substitution Principle.
      * Provides detailed fail messages according to each condition below
      * @param expenseIDAdd ExpenseID required for adding a new expense to the designated monthID MonthlyStorage Object.
      * @return ExpenseOD String message indicating success/fail add attempt.
-     * @throws EntityException thrown when expenseIDAdd has invalid expense information.
      */
     @Override
-    public ExpenseOD addExpenseInMonth(ExpenseID expenseIDAdd) throws EntityException {
-        this.expenseID = expenseIDAdd;
-        SessionStorage session = expenseIDAdd.getSession();
-        MonthlyStorage month = expenseIDAdd.getSession().getMonthlyData(expenseIDAdd.getMonthID());
-        ArrayList<Expense> monthExpenseList = month.getExpenseData();
-        ArrayList<Category> categoryList = month.getCategoryData();
-
-        try {double valueDouble = toDouble(expenseIDAdd.getValue());
-            Category selectedCategory = findCategory(categoryList, expenseID.getCategory());
+    public ExpenseOD addExpenseInMonth(ExpenseID expenseIDAdd){
+        try {SessionStorage session = expenseIDAdd.getSession();
+            this.month = session.getMonthlyData(expenseIDAdd.getMonthID());
+            double valueDouble = toDouble(expenseIDAdd.getValue());
+
             if (valueDouble < 0) {
                 //Expense value less than 0 fail: When a user tries to add the expense value with a negative number.
                 ExpenseOD expenseODFailAdd = new ExpenseOD("Expense value can't be less than $0. Please try again!");
                 return expenseOB.fail(expenseODFailAdd);}
 
-            for (Expense expense1 : monthExpenseList) {
-                if (expense1.getName().equals(expenseID.getName())) {
+            for (Expense expense1 : month.getExpenseData()) {
+                if (expense1.getName().equals(expenseIDAdd.getName())) {
```

Comment:
Why aren't you using your helper method for checking this?

Reply:
Also, I just realized that the helper method isNameInList is buggy. Inside its for loop it returns the first boolean that it checks instead of having an accumulator

Reply:
Here's a quick suggestion on how to fix it
```java
    private boolean checkHaveSameNameInList(ArrayList<Expense> expenseList, String name) {
        for (Expense expense1 : expenseList) {
            if (expense1.getName().equals(name)) {
                return true;
            }
        }
        return false;
    }
```

Reply:
Maybe you can change the method name to isExpenseNameInList. A bit shorter and I think it could be clearer on what it does

Reply:
Oh I did not realise, this is such a bug, thanks Ari!
And I will fix the name as well.